child_sync_typo3/wizard/delegate_child_wizard.py

```diff
@@ -0,0 +1,34 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: David Coninckx <david@coninckx.com>
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+from openerp.osv import orm
+from ..model.sync_typo3 import Sync_typo3
+
+
+class delegate_child_wizard(orm.TransientModel):
+    _inherit = 'delegate.child.wizard'
+
+    def delegate(self, cr, uid, ids, context=None):
+        res = super(delegate_child_wizard, self).delegate(
+            cr, uid, ids, context)
+
+        child_ids = self._default_child_ids(cr, uid, context)
+        child_obj = self.pool.get('compassion.child')
+
+        typo3_to_remove_ids = list()
+        for child in child_obj.browse(cr, uid, child_ids, context):
+            if (child.state == 'I'):

```

Comment:
This is not working as well because children will already be in delegated state. You should perform it before the super call.